File: solutions/python/all-your-base/1/all_your_base.py
def rebase(input_base, digits, output_base):
    if input_base < 2:
        raise ValueError("input base must be >= 2")

    if output_base < 2:
        raise ValueError("output base must be >= 2")

    if any([d < 0 or d >= input_base for d in digits]):
        raise ValueError("all digits must satisfy 0 <= d < input base")

    base_ten = sum([d * (input_base ** i) for i, d in enumerate(digits[::-1])])

    print(base_ten)

    out_digits = []
    while base_ten > 0:
        out_digits = [base_ten % output_base] + out_digits
        base_ten //= output_base

    return out_digits or [0]

File: solutions/r/all-your-base/1/all-your-base.R
rebase <- function(from_base, digits, to_base) {
  stopifnot(
    "input base must be >= 2" = from_base >= 2,
    "output base must be >= 2" = to_base >= 2,
    "all digits must satisfy 0 <= d < input base" = all(digits >= 0) & all(digits < from_base)
  )
  
  if (length(digits) == 0 | all(digits == 0)) { return(0) }
  
  num <- sum(digits * from_base ^ (length(digits):1-1))
  
  rebased <- c()
  while (num > 0) {
    rebased <- c(num %% to_base, rebased)
    num <- num %/% to_base
  }
  
  rebased
}
